polkadot/node/network/availability-distribution/src/requester/mod.rs

```diff
@@ -146,32 +206,196 @@ impl Requester {
 
 		// Also spawn or bump tasks for candidates in ancestry in the same session.
 		for hash in std::iter::once(leaf).chain(ancestors_in_session) {
-			let cores = get_occupied_cores(sender, hash).await?;
+			let occupied_cores = get_occupied_cores(sender, hash).await?;
 			gum::trace!(
 				target: LOG_TARGET,
-				occupied_cores = ?cores,
+				occupied_cores = ?occupied_cores,
 				"Query occupied core"
 			);
+
+			let cores = occupied_cores
+				.into_iter()
+				.map(|(index, occ)| {
+					(
+						index,
+						CoreInfo {
+							candidate_hash: occ.candidate_hash,
+							relay_parent: occ.candidate_descriptor.relay_parent(),
+							erasure_root: occ.candidate_descriptor.erasure_root(),
+							group_responsible: occ.group_responsible,
+						},
+					)
+				})
+				.collect::<Vec<_>>();
+
 			// Important:
 			// We mark the whole ancestry as live in the **leaf** hash, so we don't need to track
 			// any tasks separately.
 			//
 			// The next time the subsystem receives leaf update, some of spawned task will be bumped
 			// to be live in fresh relay parent, while some might get dropped due to the current
 			// leaf being deactivated.
-			self.add_cores(ctx, runtime, leaf, leaf_session_index, cores).await?;
+			self.add_cores(ctx, runtime, leaf, leaf_session_index, cores, FetchOrigin::Slow)
+				.await?;
 		}
 
+		let groups = get_validator_groups(sender, new_head.hash).await?;
+		if let Err(err) = self
+			.early_request_chunks(ctx, runtime, new_head, leaf_session_index, &groups)
+			.await
+		{
+			gum::warn!(
```

Comment:
We are doing a mixture of error handling here, some errors we handle with `?`, some by logging them. I would suggest to use `?` consistently and handle them at the top-level correctly - which is not the case right now (was wrong before): 

[Here](https://github.com/axaysagathiya/polkadot-sdk/blob/6dc92b47dce9ff85f0f42d72b90587484023ce9c/polkadot/node/network/availability-distribution/src/requester/mod.rs#L180) we can't just use `?`, because that will short circuit the handling of deactivated leaves. Instead we should only short circuit on fatal errors, which we can do via [log_error](https://github.com/axaysagathiya/polkadot-sdk/blob/6dc92b47dce9ff85f0f42d72b90587484023ce9c/polkadot/node/network/availability-distribution/src/error.rs#L112).